### DISCUSSION TITLE
Loading a collection results in a timeout

### DISCUSSION THREAD

**Discussion by omi-equilo** [milvus-io/milvus#28476] [Q&A and General discussion]
Using milvus v2.3.1, running as a cluster on Amazon EKS. Created a table with a vector field (dim: 768) and populated it with approx 38k entries. Index is HSNW. Schema tab screenshot is attached.

When I try to load this collection either via Attu or a command line script, I get a timeout error (60000ms). Can you help me figure out what is going wrong here?

querynode and querycoord pod logs are attached.

Thanks for the help!

[querynode-logs.txt](https://github.com/milvus-io/milvus/files/13371542/querynode-logs.txt)
[querycoord-logs.txt](https://github.com/milvus-io/milvus/files/13371543/querycoord-logs.txt)
![Screenshot 2023-11-15 at 6 25 07 PM](https://github.com/milvus-io/milvus/assets/138073801/c901de14-4b67-4f2a-851e-0e610b610b1f)

**Comment by xiaofan-luan** [comment]
Could you check the states of the collection on attu?
The timeout doesn't really mean load failed. Simply becasue attu set a 10 minutes timeout

**Reply by omi-equilo** [reply]
@xiaofan-luan The collection doesn't in fact ever load. Even if I refresh / login again to attu an hour or a day later.

**Comment by yhmo** [comment]
There are lots of these logs in querynode:
```
[2023/11/15 18:06:45.088 +00:00] [WARN] [pipeline/pipeline.go:50] ["some node(s) haven't received input"] [list="[nodeCtxTtChecker-FilterNode-by-dev-rootcoord-dml_1_444582924789417508v0,nodeCtxTtChecker-InsertNode-by-dev-rootcoord-dml_1_444582924789417508v0,nodeCtxTtChecker-DeleteNode-by-dev-rootcoord-dml_1_444582924789417508v0]"] ["duration "=2m0s]
[2023/11/15 18:08:45.088 +00:00] [WARN] [pipeline/pipeline.go:50] ["some node(s) haven't received input"] [list="[nodeCtxTtChecker-FilterNode-by-dev-rootcoord-dml_1_444582924789417508v0,nodeCtxTtChecker-InsertNode-by-dev-rootcoord-dml_1_444582924789417508v0,nodeCtxTtChecker-DeleteNode-by-dev-rootcoord-dml_1_444582924789417508v0]"] ["duration "=2m0s]
```

And the querycoord could not connect to the querynode for deadlineExceeded:
`[2023/11/16 02:14:17.411 +00:00] [WARN] [grpcclient/client.go:486] ["ClientBase Call grpc call get error"] [role=querynode-37] [address=192.168.73.161:21123] [error="stack trace: /go/src/github.com/milvus-io/milvus/pkg/tracer/stack_trace.go:51 github.com/milvus-io/milvus/pkg/tracer.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:485 github.com/milvus-
......
context done during sleep after run#0\nWraps: (6) context deadline exceeded\nError types: (1) *withstack.withStack (2) *errutil.withPrefix (3) merr.multiErrors (4) *withstack.withStack (5) *errutil.withPrefix (6) context.deadlineExceededError"]`

Similar to [this issue](https://github.com/milvus-io/milvus/issues/23238) which was fixed by [this pr](https://github.com/milvus-io/milvus/pull/23293) by enlarging a configuration value:
```
dataCoord:
  channel:
    watchTimeoutInterval: 300
```

@omi-equilo 
Some questions:
- how many partitions in the collection?
- did you change the dataCoord.channel.watchTimeoutInterval value in the milvus.yaml?

**Reply by omi-equilo** [reply]
@yhmo Only a single default partition in the collection. I did not change the dataCoord.channel.watchTimeoutInterval in my milvus.yaml. I followed the instructions at https://milvus.io/docs/eks.md, and then the only change I made after that was to enable common.security.authorizationEnabled. Could that have made a difference??

Thanks for the help on this, folks!

**Comment by xiaofan-luan** [comment]
@omi-equilo 
it seems your mq is stucked for some reason.
Querynode can you read from mq for a while.

**Reply by omi-equilo** [reply]
I don't quite follow, @xiaofan-luan. What can I try to do to fix or get around this issue?

**Reply by xiaofan-luan** [reply]
1. did you check if you kafka/pulsar is working as expected?
2. if you want to recover, restart the cluster might help

**Reply by xiaofan-luan** [reply]
currently  you only offered query side log, which shows it can not receive any notification from log storage.
My guess is that the pulsar/kafka is failed due to some reason. possibly lack of disk space

**Comment by omi-equilo** [comment]
Made some progress! From scanning through various logs I discovered that the indexnode process was in a crash loop. Logs are here.

[indexnode-logs.txt](https://github.com/milvus-io/milvus/files/13396829/indexnode-logs.txt)

Then I rebooted the nodes in this cluster, and was able to trigger the collection to load! The loading however got stuck at 33% loaded (from attu). :-( 

Eventually I saw the indexnode crashing again. Here are the logs after the restart.

[indexnode-logs-post-restart.txt](https://github.com/milvus-io/milvus/files/13396882/indexnode-logs-post-restart.txt)

Does this shed any light on whats going on here??

**Comment by xiaofan-luan** [comment]
From your original logs, your indexnode can not access object storage you are using.
There are multiple reasons of why this could happen:
For example: https://github.com/milvus-io/milvus/discussions/25620
Check GetObjectSize on github, you will find a bunch of different issue ，most common reasons including:
1. SSL and other configs setup
2. use virtual host()
3. on AWS, you may need to provide region AWS_REGION

**Reply by xiaofan-luan** [reply]
I don't believe restart node may solve these problem.

Querynode and Indexnode both rely on S3 access.
Check GetObjectSize as a keyword see if you still hitting similar issue
